rclcpp/test/rclcpp/CMakeLists.txt

```diff
@@ -358,6 +358,22 @@ if(TARGET test_qos)
     ${PROJECT_NAME}
   )
 endif()
+function(test_pubsub_for_rmw_implementation)
+  set(rmw_implementation_env_var RMW_IMPLEMENTATION=${rmw_implementation})
+  ament_add_gmock(test_pubsub${target_suffix} generic/test_pubsub.cpp
+    ENV ${rmw_implementation_env_var}
+  )
+  if(TARGET test_pubsub${target_suffix})
+    target_link_libraries(test_pubsub${target_suffix} ${PROJECT_NAME})
+    target_include_directories(test_pubsub${target_suffix} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/rclcpp/src/generic>)
```

Comment:
This `BUILD_INTERFACE` directory doesn't seem right to me, first why does it need the src folder to be on the include path? and second the path `${CMAKE_CURRENT_SOURCE_DIR}/rclcpp/src` seems wrong to me shouldn't it be `${CMAKE_CURRENT_SOURCE_DIR}/src/rclcpp`?

Reply:
Right, removed it.